Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/WorkspaceController.java

```diff
@@ -94,19 +99,43 @@ public String getWorkspaceDetails(@PathVariable("workspaceid") String workspacei
             throw new Quadriga404Exception("Workspace with ID " + workspaceid + " does not exist.");
         }
 
+        String projectId = wsManager.getProjectIdFromWorkspaceId(workspaceid);
+        IProject project = projectManager.getProjectDetails(projectId);
+
         // retrieve the collaborators associated with the workspace
         List<IWorkspaceCollaborator> workspaceCollaboratorList = workspace.getWorkspaceCollaborators();
-        List<ITextFile> tfList = tfManager.retrieveTextFiles(workspaceid);
         workspace.setWorkspaceCollaborators(workspaceCollaboratorList);
+
+        List<ITextFile> tfList = tfManager.retrieveTextFiles(workspaceid);
+
         List<IWorkspaceNetwork> workspaceNetworkList = wsManager.getWorkspaceNetworkList(workspaceid);
+
+        List<IProjectCollaborator> projectCollaborators = projectCollaboratorManager.getProjectCollaborators(projectId);
+
+        List<IProjectCollaborator> projectAdmins = new ArrayList<IProjectCollaborator>();
+
+        for (IProjectCollaborator collaborator : projectCollaborators) {
+            List<IQuadrigaRole> collaboratorRoles = collaborator.getCollaborator().getCollaboratorRoles();
+
+            List<String> roleIds = new ArrayList<String>();
+            collaboratorRoles.forEach(collaboratorRole -> roleIds.add(collaboratorRole.getId()));
+
+            if (roleIds.contains(RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN)) {
+                projectAdmins.add(collaborator);
+            }
+        }
+
+        model.addAttribute("projectAdmins", projectAdmins);
+        model.addAttribute("projectOwner", project.getOwner());
         model.addAttribute("networkList", workspaceNetworkList);
         model.addAttribute("workspacedetails", workspace);
         model.addAttribute("textFileList", tfList);
 
-        String adminRoles[] = {RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN, RoleNames.ROLE_COLLABORATOR_OWNER}; 
-        
+        String adminRoles[] = { RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN, RoleNames.ROLE_COLLABORATOR_OWNER,
+                RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN };
```

Comment:
Have you looked into the authorization check code? Looking at it, just the workspace roles should actually be enough. I forgot that I added those checks to the workspace manager.